### DISCUSSION TITLE
NATS storage problem

### DISCUSSION THREAD

**Discussion by bautistamad** [nats-io/nats-server#3764] [General]
Hi, I am using the Interest policy of nats in kubernetes with 3 replicas, and I have the following problem:
What is indicated in the stream is not what is actually stored on disk, the disk has much more capacity, I do not know if it is because the delated messages are stored, and the problem I have now is that I am filling the storage very quickly without knowing why.
![image](https://user-images.githubusercontent.com/75149705/210838775-a10e1436-1cb4-487f-a188-be52b91f29ae.png)

![image](https://user-images.githubusercontent.com/75149705/210839477-d5395509-f883-40f8-a4a2-e3b575208976.png)

This is the storage of my disks
![image](https://user-images.githubusercontent.com/75149705/210839636-d82fa88d-ed23-491e-a42c-ee8e07c61cfa.png)

**Comment by derekcollison** [comment]
Can you share a tree of the storage directory with sizes so we can take a look?

Also what server version?

This might be better as an issue vs discussion.

**Comment by bautistamad** [comment]
The nats is deployed with helm, using the following version: 0.18.1.
nats-server version: 2.9.3
Each replica has its own PV with a storage of 10gb for each one. To allocate each volume we use longhorn, so the directory where it is stored is /var/lib/longhorn.

I can raise an issue if necessary.

**Comment by derekcollison** [comment]
Still would like to see a tree view of the storage directory to see what is being held.

Yes let's convert to an issue.

**Comment by bautistamad** [comment]
I will try to get the storage tree, I have already raised the issue, thank you very much.

**Comment by bautistamad** [comment]
One more question. The deleted messages shown in the image should theoretically be removed from storage as well, shouldn't they?

**Reply by derekcollison** [reply]
If they are interior they will currently be conservatively compacted. We plan on improving this for 2.10.
